students/mattmaeda/session05/comprehensions.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python3
+
+# Problems from
+# https://uwpce-pythoncert.github.io/PythonCertDevel/exercises/comprehensions_lab.html#exercise-comprehensions
+
+def count_evens(num_list):
+    # http://codingbat.com/prob/p189616
+    # Using a list comprehension, return the number of even integers in the
+    # given list.
+    return len([i for i in num_list if i % 2 == 0])
+
+
+def dict_formatter(ref_dict):
+    # dict/set problem 1
+    #
+    # Print the dict by passing it to a string format method, so that you get
+    # something like:
+    #
+    # “Chris is from Seattle, and he likes chocolate cake, mango fruit, greek
+    # salad, and lasagna pasta”
+    return "%(name)s is from %(city)s, and he likes %(cake)s cake, %(fruit)s " \
+           "fruit, %(salad)s salad and %(pasta)s pasta" % {k: v for k, v in ref_dict.items()}
+
```

Comment:
Note there is a way to do this, so that you do not have to write a dictionary comprehension for formatting. To do this you will have to use the format method instead of %.